src/redux/modules/posts.js

```diff
@@ -0,0 +1,106 @@
+import { get, put, post } from '../../utils/request';
+import url from '../../utils/url';
+import { actions as appActions } from './app';
+import { combineReducers } from 'redux';
+
+const initialState = {
+    byId: {},
+    allIds: []
+};
+
+// action types
+export const types = {
+    FETCH_ALL_POSTS: "POSTS/FETCH_ALL_POSTS"
+};
+
+// action creators
+export const actions = {
+    // get post list
+    fetchAllPosts: () => {
+        return (dispatch, getState) => {
+            if (shouldFetchAllPosts(getState())) {
+                dispatch(appActions.startRequest());
+                return get(url.getPostList()).then(data => {
+                    dispatch(appActions.finishRequest());
+                    if (!data.error) {
+                        const { posts, postsIds, authors } = convertPostsToPlain(data);
+                        dispatch(fetchAllPostsSuccess(posts, postsIds, authors));
+                    }
+                    else {
+                        dispatch(appActions.setError(data.error));
+                    }
+                });
+            }
+        }
+    }
+};
+
+// action creators - only for internal use, no need to export, so not put them into the "actions" object
```

Comment:
For actions that are only for internal use, no need to put them into the "actions" object which for exporting.